worker-core/src/main/java/com/hpe/caf/worker/core/WorkerCore.java

```diff
@@ -507,8 +504,8 @@ public void complete(final TaskInformation taskInformation, final String queue,
                 } else {
                     // **** Normal Worker ****                    
                     // A worker with an input and output queue.
-                    byte[] output = codec.serialise(responseMessage);
-                    workerQueue.publish(taskInformation, output, queue, Collections.emptyMap(), 
+                    final byte[] output = convertAndSerializeMessage(responseMessage, taskInformation);
+                    workerQueue.publish(getTaskInformation(taskInformation), output, queue, Collections.emptyMap(), 
```

Comment:
The unwrapping code is too intermingled here.  For example we are unwrapping here for calling `workerQueue.publish` but not unwrapping just above for `workerQueue.acknowledgeTask`, or for the `abandon` code.  It needs to be centralised.

Reply:
I wasn't sure that this comment was clear to I've applied it myself: 486ea3e.
The unwrapping is now done in a separate layer introduced for the purpose.  This will be easier for future developers to maintain.